html/php-pre-challenge3/index.php

```diff
@@ -1,6 +1,47 @@
 <?php
 $limit = $_GET['target'];
+if (is_string($limit) == false || $limit < 1 || preg_match('/^([1-9]\d*|0)\.(\d+)?$/', $limit) ) {
+    header('HTTP/1.1 400 Bad Request');
+    exit;
+}
 
 $dsn = 'mysql:dbname=test;host=mysql';
 $dbuser = 'test';
 $dbpassword = 'test';
+
+try {
+    $db = new PDO ($dsn, $dbuser, $dbpassword);
+} catch (PDOException $e) {
+    echo 'DB接続エラー： ' . $e->getMessage();
```

Comment:
DB接続エラー時もスクリプトを停止させましょう。

```suggestion
    http_response_code(500);
    echo json_encode('DB接続エラー： ' . $e->getMessage(), JSON_UNESCAPED_UNICODE);
    exit;
```